lib/db.js

```diff
@@ -1,115 +1,10 @@
 import { DataStream } from 'scramjet';
 import log from './logger';
 import { ignore } from './utils';
 import { cleanup } from './errors';
 
-let _pg = null;
-function getDB () {
-  let knexConfig;
-  _pg = _pg || (async () => {
-    try {
-      knexConfig = require('../knexfile-local');
-      log.info('Using connection data from knexfile-local');
-    } catch (e) {
-      knexConfig = require('../knexfile');
-      log.info('Using connection data from production knexfile');
-    }
-
-    const pg = knex(knexConfig);
-    log.info('Connecting to database.');
-
-    return pg;
-  })()
-    .then(async pg => {
-      await pg.migrate.latest(knexConfig);
-      log.info('Database connected and migrations are handled, ready to roll!');
-      return pg;
-    });
-  // Run any needed migrations and away we go
-  return _pg;
-}
-
-/**
- * Build an object that can be inserted into our database.
- * @param {object} m measurement object
- * @return {object} an object capable of being saved into the PostgreSQL database
- */
-function convertMeasurementToSQLObject (measurement, st, pg) {
-  const obj = {
-    location: measurement.location,
-    value: measurement.value,
-    unit: measurement.unit,
-    parameter: measurement.parameter,
-    country: measurement.country,
-    city: measurement.city,
-    source_name: measurement.sourceName,
-    date_utc: measurement.date.utc,
-    source_type: measurement.sourceType,
-    mobile: measurement.mobile
-  };
-
-  // Copy object JSON to the cause field
-  obj.data = Object.assign({}, measurement);
-
-  // If we have coordinates, save them with postgis
-  if (measurement.coordinates) {
-    obj.coordinates = st.geomFromText(`Point(${measurement.coordinates.longitude} ${measurement.coordinates.latitude})`, 4326);
-  }
-
-  return obj;
-}
-
-function streamRecordsToPg (stream, pg) {
-  const st = require('knex-postgis')(pg);
-
-  const table = pg('measurements')
-    .returning('location');
-
-  return stream
-    .tap()
-    .pipe(new DataStream())
-    .setOptions({maxParallel: 1})
-    .assign(async measurement => {
-      const record = convertMeasurementToSQLObject(measurement, st, pg);
-      try {
-        await table.insert(record);
-      } catch (cause) {
-        if (cause.code === '23505') {
-          return { status: 'duplicate' };
-        }
-
-        throw cause;
-      }
-      return { status: 'inserted' };
-    });
-}
-
-/**
- * Streams measurements to database and retuns stream of only the inserted items.
- *
- * @param {DataStream} stream
- * @param {Knex} pg
- * @param {Object} counts
- */
-function streamDataToDB (stream, pg, counts) {
-  return stream
-    .use(streamRecordsToPg, pg)
-    .filter(({status}) => {
-      if (status === 'duplicate') {
-        counts.duplicates++;
-        return false;
-      }
-
-      counts.inserted++;
-      return true;
-    })
-  ;
-}
-
 function saveResultsToS3 (output, s3, bucketName, key, s3ChunkSize) {
   const stream = new DataStream();
 
```

Comment:
On https://github.com/openaq/openaq-fetch/pull/797/files#diff-69f9a48e7478eb5df50c35b61f5638a078f274a604fff85d69732c2651ef8b47R13, I am not sure `status` is defined anymore? This may cause some of the data to be stripped off here, unsure.

Reply:
The `status` here is only checking the duplication in the database and it doesn't show up anywhere else. But since we're not writing to the DB anymore, it doesn't matter